feat(ui): toast

EurFelux opened the pull request:
考虑两种API形式：

1. 完全兼容现有代码的api，仍然通过将toast对象挂载到window上调用。这种方式对每个窗口都需要重新挂载一次。
2. 与[shadcn sonner](https://ui.shadcn.com/docs/components/sonner)基本相同的api调用方式，从ui库导入的toast对象暴露了`success`, `warning`, `dismiss`等方法，而`toast`本身也是可调用的函数。

目前采用的是第二种方案。

还有部分设计问题未沟通清楚，见Figma comments。

EurFelux committed:
feat(ui): add sonner toaster and next-themes support

- Add sonner package for toast notifications with custom icons and styling
- Add next-themes package for theme support
- Create new Toaster component with theme-aware styling

EurFelux committed:
feat(sonner): replace lucide icons with custom svg icons for toaster

EurFelux committed:
feat(components): export Toaster from sonner primitives

EurFelux committed:
refactor(toast): remove deprecated toast utilities and update comment

Remove deprecated toast API functions that were no-ops and update dataLimit comment to reflect current toast limitations

EurFelux committed:
feat(sonner): implement custom toast component with enhanced features

- Add custom toast component with support for info, success, warning, error, and loading states
- Implement action buttons, links, colored messages, and dismissable toasts
- Add TypeScript type definitions and Storybook documentation
- Export toast API for programmatic usage

EurFelux committed:
feat(sonner): add dismiss method to toast API

Add the ability to dismiss specific toast notifications by their ID. This provides more control over toast management in the UI.

EurFelux committed:
refactor(toast): update toast utilities and imports

- Move ToastUtilities type to local component file
- Add closeToast functionality to toast utilities
- Update error block to use success toast method

EurFelux committed:
chore: remove next-themes dependency as it's no longer needed

EurFelux committed:
fix: remove outdated toast comment and cleanup code

The toast component now supports programmatic closing, making the TODO comment obsolete. Clean up the code by removing the unnecessary comment.

EurFelux committed:
style(sonner): add todo comment for missing button styles

EurFelux committed:
build(ui): replace sonner with @cherrystudio/sonner dependency

Update package.json and yarn.lock to use @cherrystudio/sonner instead of sonner

EurFelux committed:
style(sonner): update toast background colors and add border

- Add backdrop blur effect to toast backgrounds
- Update color opacity values for better visual consistency
- Add border to toast container for better separation

EurFelux committed:
feat: add Toaster component for notifications

MyPrototypeWhat commented on packages/ui/src/components/primitives/sonner.tsx:
> [!NOTE]
> This comment was translated by Claude.

Is the icon customized by us?

---
<details>
<summary>Original Content</summary>

icon是我们自定义的吗?

</details>

MyPrototypeWhat reviewed (commented):
I don't think it's necessary to mount it on window, just call it directly after importing toast

MyPrototypeWhat commented on packages/ui/package.json:
```diff
@@ -44,6 +44,7 @@
     "tailwindcss": "^4.1.13"
   },
   "dependencies": {
+    "@cherrystudio/sonner": "2.0.8",
```
> [!NOTE]
> This issue/comment/review was translated by Claude.

What problems cannot be solved?
If implementing it ourselves, it can be placed in the UI library, right?

---
<details>
<summary>Original Content</summary>

有哪些问题是没办法解决的?
自行实现的话可以放在ui库中吧

</details>

EurFelux replied:
> [!NOTE]
> This issue/comment/review was translated by Claude.

There might also be a solution that doesn't require forking, let me experiment with it.

---
<details>
<summary>Original Content</summary>

可能也有不用fork的解决方案，等我实验一下

</details>

EurFelux replied:
> [!NOTE]
> This issue/comment/review was translated by Claude.

This doesn't look quite right.

---
<details>
<summary>Original Content</summary>

不太行的样子

</details>

EurFelux committed:
refactor(ui): replace @cherrystudio/sonner with sonner package

Update toast component to use the standard sonner package instead of the cherrystudio fork. This simplifies dependencies and ensures better maintainability with the upstream package.

EurFelux commented on src/renderer/src/utils/dataLimit.ts:
```diff
@@ -83,8 +83,7 @@ export async function checkDataLimit() {
     } else if (!shouldShowWarning && currentToastId) {
-      // Dismiss toast when space is recovered
-      window.toast.closeToast(currentToastId)
+      window.toast.close(currentToastId)
```
> [!NOTE]
> This issue/comment/review was translated by Claude.

Should this be changed in batches in this PR? It involves quite a lot.

---
<details>
<summary>Original Content</summary>

这个要在这个pr中批量改掉吗？涉及还挺多的

</details>

EurFelux commented on tsconfig.web.json:
```diff
@@ -11,6 +11,7 @@
     "packages/extension-table-plus/**/*",
     "packages/mcp-trace/**/*",
     "packages/shared/**/*",
+    "packages/ui/**/*",
```
> [!NOTE]
> This issue/comment/review was translated by Claude.

It will cause an error if not added

---
<details>
<summary>Original Content</summary>

不加会报错

</details>

MyPrototypeWhat replied:
加了会报错...ci没过是因为ui库脚本有个变量没用到

EurFelux committed:
refactor: remove ui package from tsconfig.web.json

EurFelux committed:
feat(sonner): rewrite toast to use native api

- Simplify toast API to match sonner's interface
- Add support for colored backgrounds and custom durations
- Improve icon rendering with unique IDs to prevent conflicts
- Remove unused code and consolidate styles
- Update stories to reflect new API changes

EurFelux committed:
fix(vscode): update tailwindCSS regex to include cn function

Add support for the cn function in tailwindCSS class regex to ensure proper class name detection

MyPrototypeWhat reviewed (commented):
> [!NOTE]
> This issue/comment/review was translated by Claude.

There's also the icon issue, let's wait and see which icon library to use

---
<details>
<summary>Original Content</summary>

还有icon的问题,等等看用哪个icon库

</details>

MyPrototypeWhat committed:
Merge branch 'v2' into v2-toast

EurFelux committed:
fix(toast): Update import path for cn utility in sonner component

EurFelux committed:
fix: Return toast id

EurFelux committed:
refactor: Refactor toast component with discriminated union types

EurFelux committed:
refactor: Update toast.dismiss to use ToastIdType parameter

EurFelux committed:
refactor: Update toast type definitions for improved API

- Replace `ToastType` with discriminated union `ToastProps`
- Add `BaseToastProps` for shared properties
- Introduce `CustomToastProps` with JSX render function
- Update quick API methods to accept message as first parameter
- Add `toast.custom()` method for custom toast rendering
- Export all individual toast type interfaces

EurFelux committed:
feat: Update Sonner stories with new features and examples

- Add dismissable control to playground
- Add custom toast example with JSX rendering
- Add dismissable control showcase
- Add custom class names example
- Update component description for clarity
- Import CheckIcon for custom toast example

EurFelux committed:
fix: Fix colored toast background variant selection

EurFelux committed:
fix: Fix SVG attribute names to use camelCase for React

EurFelux committed:
feat: Add close button option to toast component

- Add `closeButton` prop to BaseToastProps interface
- Update dismissable prop description for clarity
- Add close button styling to classNames
- Pass closeButton prop to external toast configuration
- Update Storybook stories to include closeButton control
- Rename "Dismissable Control" story to "Close Button Control"

EurFelux committed:
chore: Remove commented-out CloseIcon component from sonner.tsx

EurFelux committed:
fix: recover comment

EurFelux commented on packages/ui/src/components/primitives/sonner.d.ts:
类型声明文件有必要保留，因为`info`, `success`等api是在module中动态添加的，必须通过这种方式来声明。

MyPrototypeWhat replied:
这种算是静态,ts能推断出来

EurFelux committed:
Merge branch 'v2' of github.com:CherryHQ/cherry-studio into v2-toast

EurFelux committed:
build: add ui package to tsconfig.web.json includes

EurFelux committed:
refactor(ui): remove sonner type definitions as they're no longer needed

EurFelux committed:
style(sonner): change important symbol to the end

EurFelux committed:
build: update dependencies including rolldown to beta.59 and add sonner

Update multiple dependencies to their latest versions, including rolldown to beta.59 across all platforms and add the sonner package for toast notifications

EurFelux committed:
fix(sonner.stories): inline mock promise in loading toast story

EurFelux committed:
feat(sonner): add promise with callbacks story example

Demonstrate handling success/error cases in promise-based toasts with retry functionality
